staging/src/k8s.io/legacy-cloud-providers/azure/retry/azure_error.go

```diff
@@ -0,0 +1,198 @@
+// +build !providerless
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package retry
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/klog"
+)
+
+var (
+	// The function to get current time.
+	now = time.Now
+)
+
+// Error indicates an error returned by Azure APIs.
+type Error struct {
+	// Retriable indicates whether the request is retriable.
+	Retriable bool
+	// HTTPStatusCode indicates the response HTTP status code.
+	HTTPStatusCode int
+	// RetryAfter indicates the time when the request should retry after throttling.
+	// A throttled request is retriable.
+	RetryAfter time.Time
```

Comment:
`RetryAfter` is not used in this PR, right?

Reply:
Right, it would be used in future PRs.